src/theory/bv/theory_bv_rewrite_rules.h

```diff
@@ -54,6 +54,7 @@ enum RewriteRuleId
   SgtEliminate,
   RedorEliminate,
   RedandEliminate,
+  AshrEliminate,
```

Comment:
Hi Martin, thanks for the comments.
This will not be used by default.
I am writing a pre-processing pass that translates bit-vector formulas to integer formulas.
In that pass, it is convenient to assume that some operators are already eliminated, including `bvashr`.
Therefore, I would like to eliminate this operator before I reach that preprocessing pass.
Now, this elimination procedure is very similar to other elimination procedures that occur in `src/theory/bv/theory_bv_rewrite_rules_operator_elimination.h`, and this is the reason I added it to that file.

Following an offline discussion with @aniemetz , I am now wondering whether this rewrite rule should instead be implemented inside my preprocessing pass, leaving the bit-vector rewriter intact. @mpreiner, @aniemetz, @martin-cs  --  what do you think?

Reply:
If it's not on by default and currently only used in the new preprocessing pass, it should be ok. But what I actually don't like (and this is a problem with more elimination rules) is that with the ITE we introduce new Boolean terms. We should probably think of using `BITVECTOR_ITE` in this case so that everything is sent to the BV theory. @yoni206, you don't have to change that now, but this is something we will evaluate in the future for the new bit-vector solver.

Reply:
I would be happy for it to live with the other bit-vector re-write but to be called from your pass.  As previously discussed I would be concerned if it was on by default.

Reply:
Thinking of it a bit more, @martin-cs, I realized that I can even optimize this further for integer: The condition in the ITE is whether or not the msb is 0. This is done with extracts. But when translating to integers, it is much better to check this by comparison to 2^{k-1}, k being the bitwidth. So looking at this integer-oriented version of the rewrite rule, I don't think it should be in the bv rewriter, because it will never be used by the bv rewriter.